geniza/corpus/management/commands/tei_to_annotation.py

```diff
@@ -0,0 +1,347 @@
+                    self.style.WARNING(
+                        "No transcription content in %s; skipping" % xmlfile
+                    )
+                )
+                continue
+            # get the document for the file based on id / old id
+            doc = self.get_pgp_document(xmlfile_basename)
+            # if document was not found, skip
+            if not doc:
+                self.stdout.write(
+                    self.style.WARNING("Document not found for %s; skipping" % xmlfile)
+                )
+                continue
+            # found the document
+            if self.verbosity >= self.v_normal:
+                self.stdout.write(doc)
+
+            # get the footnote for this file & doc
+            footnote = self.get_edition_footnote(doc, tei, xmlfile)
+            # if no footnote, skip for now
+            # TODO: generate footnote when needed! (but should we need to?)
+            if not footnote:
+                self.stdout.write(
+                    self.style.ERROR(
+                        "footnote not found for %s / %s; skipping" % (xmlfile, doc.pk)
+                    )
+                )
+                continue
+            footnote = self.migrate_footnote(footnote, doc)
+
+            # if there is a single primary language, use the iso code if it is set
+            lang_code = None
+            if doc.languages.count() == 1:
+                lang_code = doc.languages.first().iso_code
+
+            # get html blocks from the tei
+            html_blocks = tei.text_to_html(block_format=True)
+
+            # get canvas objects for the images in order; skip any non-document images
+            iiif_canvases = [img["canvas"] for img in doc.iiif_images(filter_side=True)]
+            # determine the number of canvases needed based on labels
+            # that indicate new pages
+            # check and count any after the first; always need at least 1 canvas
+            num_canvases = 1 + len(
+                [
+                    b["label"]
+                    for b in html_blocks[1:]
+                    if tei.label_indicates_new_page(b["label"])
+                ]
+            )
+            # in verbose mode report on available/needed canvases
+            if self.verbosity > self.v_normal:
+                self.stdout.write(
+                    "%d iiif canvases; need %d canvases for %d blocks"
+                    % (len(iiif_canvases), num_canvases, len(html_blocks))
+                )
+            # if we need more canvases than we have available,
+            # generate local canvas ids
+            if num_canvases > len(iiif_canvases):
+                # document manifest url is /documents/pgpid/iiif/manifest/
+                # create canvas uris parallel to that
+                canvas_base_uri = doc.manifest_uri.replace("manifest", "canvas")
+                for i in range(num_canvases - len(iiif_canvases)):
+                    canvas_uri = "%s%d/" % (canvas_base_uri, i + 1)
```

Comment:
Ah, great! I see the `i + 1` here now, didn't read this section carefully enough 😄